retriever/lib/engine.py

```diff
@@ -60,11 +61,12 @@ def disconnect(self):
             self.connection.close()
             self._connection = None
             self._cursor = None
-
+    
+    
     def get_connection(self):
         """This method should be overloaded by specific implementations
         of Engine."""
-        pass
+        raise NotImplementedError
```

Comment:
In Python,  it's not required to implement the abstract methods of the parent class. Unless we really want that particular method to be implemented by any inheriting class, otherwise the previous version was fine.

Reply:
In other languages you have to implement the abstract methods

Reply:
So is this pull request not required?

Reply:
I think this small change can be included for engine.py which might be needed at a later stage.